Sources/TestSupport/MockDependencyResolver.swift

```diff
@@ -124,6 +125,17 @@ public struct MockPackagesProvider: PackageContainerProvider {
         }
         throw MockLoadingError.unknownModule
     }
+
+    public func getContainer(for identifier: Container.Identifier, completion: @escaping (Result<Container, AnyError>) -> Void) {
+        requestedContainers.append(identifier)
+        let result: Result<Container, AnyError>
+        do {
+            result = Result(try getContainer(for: identifier))
+        } catch {
+            result = Result(AnyError(error))
+        }
+        completion(result)
```

Comment:
Let's make this behave asynchronously (via an internal parallel queue) just to ensure we are getting test coverage of that behavior.